src/main/java/de/undercouch/gradle/tasks/download/DownloadAction.java

```diff
@@ -205,23 +205,27 @@ private void execute(URL src, Project project) throws IOException {
      * server if the given timestamp is greater than 0.
      * @param src the source URL to open a connection for
      * @param timestamp the timestamp of the destination file
+     * @param length the length of the destination file
      * @param project the project to be built
      * @return the URLConnection or null if the download should be skipped
      * @throws IOException if the connection could not be opened
      */
-    private URLConnection openConnection(URL src, long timestamp,
+    private URLConnection openConnection(URL src, long timestamp, long length,
             Project project) throws IOException {
         int redirects = MAX_NUMBER_OF_REDIRECTS;
         
         URLConnection uc = src.openConnection();
         while (true) {
-            if (uc instanceof HttpURLConnection) {
+            long contentLength = parseContentLength(uc);
+            uc = uc.getURL().openConnection();
```

Comment:
Wouldn't it be better to do the whole thing after line 160 where the content length is requested anyhow? Just guessing. I need to check this more in detail, but I guess it would make the code a lot more cleaner. WDYT?